internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -230,6 +230,18 @@ func (r *Reconciler) reconcileControlPlane(ctx context.Context, s *scope.Scope)
 		return err
 	}
 
+	// If the controlPlane has infrastructureMachines and the InfrastructureMachineTemplate has changed on this reconcile
+	// delete the old template.
+	// This is a best effort deletion only and may leak templates if an error occurs during reconciliation.
```

Comment:
Are there any reasons to keep a revision history here? I think some of the core types that have child/dependents like this keep the last few around and then GC them only once they are a certain age/are a number of revisions out of date

Reply:
I don't think so - the templates here are copied from a base type which is the source of truth for the spec of the ControlPlane machine so maybe we would want some history for the underlying object. 

This change is just a best effort deletion to stop the resource leaking though - I think we need a more holistic solution to cleaning up these resources and maybe we should think about whether a revision history is something we'd like to support.

Reply:
Seems like a reasonable argument to me, feel free to resolve this